react-paw-mailmerge/src/Helpers/CsvFunctions.tsx

```diff
@@ -92,4 +92,62 @@ function NormalizeJsonObjectResult(parsedCsv:ParseResult<object>, options:CsvOpt
    return result;
 }
 
+export function CheckTemplate(templateMessage: string, headers:Array<string>|undefined):Array<string>{
+    let errorMessages = new Array<string>();
+    console.log("Headers: ", headers);
+    var fields = GetFields(templateMessage);
+    for(const i in fields){
+        console.log("FiledName: ", i, fields[i]);
+        if(headers === null || !headers?.includes(fields[i]?.replace("{{", "").replace("}}",""))){
+            errorMessages.push(`Undefined Field - ${fields[i]}`);
+        }
+    }
+
+    //Do we have any {{ or }} values that have not been identified as fields?
+    const dummyObj = {}
+    let modifiedMessage = DoMailMerge(dummyObj, templateMessage);
+    // all fields are going to be undefined because we've passed in an empty object
+    modifiedMessage = modifiedMessage.replaceAll(UNDEFINED_FIELD, ""); 
+    const doubleOpen = modifiedMessage.match(/{{/g);
+    const doubleClose = modifiedMessage.match(/}}/g);
+
+    if( (doubleOpen?.length ?? 0) !== 0){
+        errorMessages.push("Invalid Syntax -- Opening double braces ('{{') without appropriate closing double braces ('}}')");
+    }
+
+    if((doubleClose?.length ?? 0) !== 0){
+        errorMessages.push("Invalid Syntax -- Closing double braces ('}}') without appropriate opening double braces ('{{')");
+    }
+
+    return errorMessages;
+}
+
+export function DoMailMerge(rowObj: any, templateMessage: string):string{
+    var fields = GetFields(templateMessage);
+    rowObj = rowObj ?? {};
```

Comment:
It's probably safe to assume that `rowObj` won't be null or undefined. The caller should always supply an object.